src/transformers/models/seamless_m4t/modeling_seamless_m4t.py

```diff
@@ -0,0 +1,4058 @@
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        **kwargs,
+    ) -> Union[Seq2SeqLMOutput, Tuple[torch.FloatTensor]]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Labels for computing the masked language modeling loss. Indices should either be in `[0, ...,
+            config.vocab_size]` or -100 (see `input_ids` docstring). Tokens with indices set to `-100` are ignored
+            (masked), the loss is only computed for the tokens with labels in `[0, ..., config.vocab_size]`.
+
+        Returns:
+
+        """
+
+        logger.warning(
+            "This calls `self.input_model.forward`. If you want to generate speech, use the `generate` method."
+        )
+
+        return self.input_model.forward(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            head_mask=head_mask,
+            decoder_head_mask=decoder_head_mask,
+            cross_attn_head_mask=cross_attn_head_mask,
+            encoder_outputs=encoder_outputs,
+            past_key_values=past_key_values,
+            inputs_embeds=inputs_embeds,
+            decoder_inputs_embeds=decoder_inputs_embeds,
+            labels=labels,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            **kwargs,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        **kwargs,
+    ) -> Union[str, torch.LongTensor]:  # TODO: output
+        kwargs_text_generation = {}
+        kwargs_speech_generation = {}
+        for key, value in kwargs.items():
+            if key.startswith("text_generation_"):
+                key = key[len("text_generation_") :]
+                kwargs_text_generation[key] = value
+            elif key.startswith("speech_generation_"):
+                key = key[len("speech_generation_") :]
+                kwargs_speech_generation[key] = value
+            else:
+                # If the key is already in a specific config, then it's been set with a
+                # submodules specific value and we don't override
+                if key not in kwargs_text_generation:
+                    kwargs_text_generation[key] = value
+                if key not in kwargs_speech_generation:
+                    kwargs_speech_generation[key] = value
+
+        kwargs_text_generation["output_hidden_states"] = True
+        kwargs_text_generation["return_dict_in_generate"] = True
+        kwargs_text_generation["output_scores"] = True
+
+        output_text = self.input_model.generate(input_ids, **kwargs_text_generation)
```

Comment:
This is going to use the generation config for the `input_model` right? So we should define the correct `generation_config` for each of these sub-models?

Reply:
Would say `generation_outputs` is a better naming here

Reply:
This is a really good question that need to be solved.

@gante , we would love to have your thoughts on this as well!